src/main/java/com/segment/analytics/android/integrations/amplitude/AmplitudeIntegration.java

```diff
@@ -321,14 +320,13 @@ private void event(
   }
 
   private boolean getOptOutOfSessionFromOptions(@Nullable Map options) {
-    boolean outOfSession = false;
-
-    if (!isNullOrEmpty(options)) {
-      if (options.containsKey("outOfSession") && options.get("outOfSession") instanceof Boolean) {
-        outOfSession = (Boolean) options.get("outOfSession");
-      }
+    if (isNullOrEmpty(options)) {
+      return false;
+    }
+    if (options.containsKey("outOfSession") && options.get("outOfSession") instanceof Boolean) {
```

Comment:
These two can also be it's own if checks and it could be simplified.

Pseudocode-ish:
```
Object outOfSession = options.get("outOfSession");
if outOfSession == null {
  return false
}
if outOfSession instanceOf Boolean {
  return (boolean) outOfSession;
}
return false;
```

The reason for doing it this way is it's clearer to see all the cases you have and makes it easier to write tests for all these cases , especially if you're using a code coverage tool to see what you have missed.